tests/datastore/test_base.py

```diff
@@ -264,3 +264,82 @@ def test_as_df_time_filters(start_time_tz, end_time_tz, df_tz):
+    Tests filtering data from 2020-2022 with year=YYYY/month=MM partitions.
+    """
+    time_column = "timestamp"
+
+    # Create test data spanning 2020-2022, all months
+    test_data = []
+    for year in range(2020, 2023):  # 2020, 2021, 2022
+        for month in range(1, 13):  # Jan-Dec
+            for day in [5, 15, 25]:  # 3 records per month
+                timestamp = datetime(year, month, day, 12, 0, 0)
+                test_data.append(
+                    {
+                        "timestamp": timestamp,
+                        "year": year,
+                        "month": month,
+                        "value": year * 100 + month * 10 + day,
+                    }
+                )
+
+    df = pd.DataFrame(test_data)
+
+    # Create partitioned parquet structure: year=YYYY/month=MM/
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Write partitioned parquet files
+        df.to_parquet(temp_dir, partition_cols=["year", "month"], engine="pyarrow")
+
+        # Create DataItem with trailing slash for directory
+        dir_url = f"file://{temp_dir}/"
+        data_item = mlrun.datastore.store_manager.object(dir_url)
+
+        # Test filter: Get data from June 2021 to August 2021
+        start_time = datetime(2021, 6, 10, 0, 0, 0)  # June 10, 2021
+        end_time = datetime(2021, 8, 20, 0, 0, 0)  # August 20, 2021
+
+        # Execute the filter
+        result = data_item.as_df(
+            start_time=start_time,
+            end_time=end_time,
+            time_column=time_column,
+            format="parquet",
+        )
+
+        # Calculate expected results manually
+        # Should include: June 15&25, July 5&15&25, August 5&15 = 7 records
+        expected_df = df[(df[time_column] > start_time) & (df[time_column] <= end_time)]
+
+        expected_count = 7  # June(2) + July(3) + August(2) = 7 records
+
+        # Verify results
+        assert len(result) == expected_count, (
+            f"Expected {expected_count} rows, got {len(result)} rows. "
+            f"Filter: {start_time} to {end_time}"
+        )
+
+        assert len(result) == len(
+            expected_df
+        ), f"Expected {len(expected_df)} rows (calculated), got {len(result)} rows"
+
+        # Verify all results are within the expected date range
+        for _, row in result.iterrows():
+            row_date = pd.to_datetime(row["timestamp"])
+            assert (
+                row_date > start_time
+            ), f"Date {row_date} should be after {start_time}"
+            assert (
+                row_date <= end_time
+            ), f"Date {row_date} should be before/equal {end_time}"
+
+        # Verify we got data from the right months
+        result_months = set(pd.to_datetime(result["timestamp"]).dt.month)
+        expected_months = {6, 7, 8}  # June, July, August
+        assert (
+            result_months == expected_months
+        ), f"Expected months {expected_months}, got {result_months}"
```

Comment:
Shorter and clearer to just compare the dataframes (+ improved comment a bit):
```suggestion
        # Should include: June 15 & 25 + July 5, 15 & 25 + August 5 & 15 = 7 records
        expected_df = df[
            (df[time_column] > start_time) & (df[time_column] <= end_time)
        ].reset_index(drop=True)

        pd.testing.assert_frame_equal(
            result,
            expected_df,
            check_like=True,
            check_dtype=False,
            check_categorical=False,
        )
```